### PR TITLE
Update instagram_pic.py

### PR DESCRIPTION
**Pull request by nababuddin**
### Describe your change:

* [X] Add an algorithm?
* [X] Fix a bug or typo in an existing algorithm?
* [ ] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [ ] Documentation change?

### Checklist:
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [X] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [X] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Review by nababuddin — COMMENTED** [review]
Fixed the issues

**Review by nababuddin — COMMENTED** [review]
Review Please

**Review by nababuddin — COMMENTED** [review]
Fixed the issues

File: web_programming/instagram_pic.py [hunk 1]
```diff
@@ -1,16 +1,37 @@
...
+                        file.write(image_data)
+                    print(f"Image downloaded and saved as {file_name}")
+                else:
+                    print("Failed to download the image.")
+            else:
+                print("Image URL not found in the meta tag.")
+        else:
+            print("No meta tag with 'og:image' property found.")
+    except requests.exceptions.RequestException as e:
+        print(f"An error occurred during the HTTP request: {e}")
```
Comment (cclauss) [inline web_programming/instagram_pic.py:31]: CONTRIBUTING.md: Algorithmic functions should not `print()`.

File: web_programming/instagram_pic.py [hunk 2]
```diff
@@ -1,37 +1,68 @@
 from datetime import datetime
+from typing import Optional
```
Comment (cclauss) [inline web_programming/instagram_pic.py]: Not needed on Python 3.12
```suggestion

```

Comment (nababuddin) [reply web_programming/instagram_pic.py]: OK

Comment (nababuddin) [reply web_programming/instagram_pic.py]: @cclauss tried many times but every time getting this error
Error: web_programming/instagram_pic.py:1:1: I001 Import block is un-sorted or un-formatted

Comment (cclauss) [reply web_programming/instagram_pic.py]: If the imports were working fine before, then do not change them.

Click the text __I001__ to read the rule and how it works.

`python3 -m pip install ruff` and then `ruff --select=I --fix web_programming/instagram_pic.py`

**Review by cclauss — APPROVED** [review]

**Pull request merged**